File: src/main/java/popa/catalin/lab1/domain/Ingredient.java
package popa.catalin.lab1.domain;

import java.util.Objects;

public class Ingredient {
    private Float quantity;
    private String unit;
    private String name;
    private String adjective;

    public Ingredient(Float quantity, String unit, String name) {
        this.quantity = quantity;
        this.unit = unit;
        this.name = name;
    }

    public Ingredient(Float quantity, String unit, String name, String adjective) {
        this.quantity = quantity;
        this.unit = unit;
        this.name = name;
        this.adjective = adjective;
    }

    public Float getQuantity() {
        return quantity;
    }

    public void setQuantity(Float quantity) {
        this.quantity = quantity;
    }

    public String getUnit() {
        return unit;
    }

    public void setUnit(String unit) {
        this.unit = unit;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getAdjective() {
        return adjective;
    }

    public void setAdjective(String adjective) {
        this.adjective = adjective;
    }

    @Override
    public String toString() {
        if (adjective != null)
            return "Ingredient{" +
                    "quantity=" + quantity +
                    ", unit='" + unit + '\'' +
                    ", name='" + name + '\'' +
                    ", adjective='" + adjective + '\'' +
                    '}';
        else
            return "Ingredient{" +
                    "quantity=" + quantity +
                    ", unit='" + unit + '\'' +
                    ", name='" + name + '\'' +
                    '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Ingredient)) return false;
        Ingredient that = (Ingredient) o;
        return quantity.equals(that.quantity) &&
                unit.equals(that.unit) &&
                name.equals(that.name) &&
                Objects.equals(adjective, that.adjective);
    }

    @Override
    public int hashCode() {
        return Objects.hash(quantity, unit, name, adjective);
    }
}

File: LabsSimonaBurlacu232/Lab1/Lab1/Lab1/Domain/Ingredient.cs
namespace Lab1.Domain
{
    public class Ingredient
    {
        private double quantity;
        private Unit unit;
        public IngredientType ingredientType;
        public CoffeeFeature? coffeFeature;

        public Ingredient(double quantity, Unit unit, IngredientType ingredient, CoffeeFeature? coffeFeature)
        {
            this.quantity = quantity;
            this.unit = unit;
            this.ingredientType = ingredient;
            this.coffeFeature = coffeFeature;
        }

        public bool IsSameIngredient(Ingredient b)
        {
            var a = this;
            if (b == null) return false;
            return a.quantity == b.quantity && a.coffeFeature == b.coffeFeature && a.ingredientType == b.ingredientType && a.unit == b.unit;
        }
    }
}
